bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/SetpointRenderer.java

```diff
@@ -79,29 +70,11 @@ public boolean canRender(Widget w) {
             maxValue = BigDecimal.valueOf(100);
         }
 
-        // if the current state is a valid value, we calculate the up and down step values
-        if (state instanceof DecimalType) {
-            DecimalType actState = (DecimalType) state;
-            BigDecimal newLower = actState.toBigDecimal().subtract(step);
-            BigDecimal newHigher = actState.toBigDecimal().add(step);
-            if (newLower.compareTo(minValue) < 0) {
-                newLower = minValue;
-            }
-            if (newHigher.compareTo(maxValue) > 0) {
-                newHigher = maxValue;
-            }
-            newLowerState = newLower.toString();
-            newHigherState = newHigher.toString();
-        }
-
         String unit = getUnitForWidget(w);
 
-        String snippetName = "setpoint";
-        String snippet = getSnippet(snippetName);
+        String snippet = getSnippet("setpoint");
 
         snippet = preprocessSnippet(snippet, w);
-        snippet = StringUtils.replace(snippet, "%newlowerstate%", newLowerState);
-        snippet = StringUtils.replace(snippet, "%newhigherstate%", newHigherState);
```

Comment:
The snippet doesn't contain `%newlowerstate%` or `%newhigherstate%` so all logic for this can be removed.